modules/dhcp/dhcp_api.rb

```diff
@@ -95,17 +131,52 @@ def load_subnet_data
     end
   end
 
-  # delete a record from a network
+  # deprecated, delete a record from a network
   delete "/:network/:record" do
     begin
       load_subnet
       load_subnet_data
 
+      logger.warn('DELETE dhcp/:network/:record endpoint has been deprecated and will be removed in future versions. '\
+                  'Please use DELETE dhcp/:network/mac/:mac_address or DELETE dhcp/:network/ip/:ip_address instead.')
+
       record = server.find_record(@subnet.network, params[:record])
-      log_halt 404, "DHCP record #{params[:network]}/#{params[:record]} not found" unless record
-      server.del_record @subnet, record
+      log_halt 404, "No DHCP record for #{params[:network]}/#{params[:record]} found" unless record
+      server.del_record(@subnet, record).to_json
+    rescue Proxy::DHCP::InvalidRecord
+      log_halt 404, "No DHCP records for #{params[:network]}/#{params[:record]} found"
+    rescue Exception => e
+      log_halt 400, e
+    end
+  end
+
+  # deletes all records for an ip address from a network
+  delete "/:network/ip/:ip_address" do
+    begin
+      load_subnet
+      load_subnet_data
+
+      records = server.find_records_by_ip(@subnet.network, params[:ip_address])
+      log_halt 404, "No DHCP records for #{params[:network]}/#{params[:ip_address]} found" unless records
+      records.map { |record| server.del_record @subnet, record }.to_json
```

Comment:
Yes, that's why I was asking. I'd suggest removing the 404 completely for the time being.